x/did/types/document_sign.go

```diff
@@ -0,0 +1,152 @@
+	ariesdid "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/ld"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/jsonld"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/signer"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite/ecdsasecp256k1signature2019"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/context"
+	ldstore "github.com/hyperledger/aries-framework-go/pkg/store/ld"
+	jsld "github.com/piprate/json-gold/ld"
+)
+
+func SignDocument(doc []byte, verificationID string, sequence uint64, privKey *btcec.PrivateKey) ([]byte, error) {
+
+	signerContext := NewSignerContext(verificationID, ES256K_2019_SIG, sequence)
+	documentSigner := GetECDSASigner(privKey)
+	loader, err := GetDocumentLoader()
+	if err != nil {
+		return nil, fmt.Errorf("error to get document loader: %v", err)
+	}
+
+	signedDocument, err := documentSigner.Sign(signerContext, doc, jsonld.WithDocumentLoader(loader))
+	if err != nil {
+		return nil, fmt.Errorf("error to sign document: %v", err)
+	}
+
+	return signedDocument, nil
+}
+
+func VerifyProof(doc ariesdid.Doc) error {
+
+	sigVerifier := ecdsasecp256k1signature2019.New(suite.WithVerifier(ecdsasecp256k1signature2019.NewPublicKeyVerifier()))
+	loader, err := GetDocumentLoader()
+	if err != nil {
+		return err
+	}
+	if err := doc.VerifyProof([]verifier.SignatureSuite{sigVerifier}, jsonld.WithDocumentLoader(loader)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func GetDocumentLoader() (*ld.DocumentLoader, error) {
+	storeProvider := mem.NewProvider()
+	contextStore, err := ldstore.NewContextStore(storeProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	remoteProviderStore, err := ldstore.NewRemoteProviderStore(storeProvider)
+	if err != nil {
+		return nil, err
+	}
+	ctx, err := context.New(
+		context.WithJSONLDContextStore(contextStore),
+		context.WithJSONLDRemoteProviderStore(remoteProviderStore),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	loader, err := ld.NewDocumentLoader(
+		ctx,
+		ld.WithRemoteDocumentLoader(&httpRemoteDocumentLoader{}),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return loader, nil
+}
+
+type httpRemoteDocumentLoader struct{}
+
+func (l *httpRemoteDocumentLoader) LoadDocument(url string) (*jsld.RemoteDocument, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	doc, err := jsld.DocumentFromReader(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return &jsld.RemoteDocument{
+		DocumentURL: url,
+		Document:    doc,
+	}, nil
+}
+
+func NewSignerContext(vmID string, signatureType string, sequence uint64) *signer.Context {
+	return &signer.Context{
+		SignatureType: signatureType,
+		Creator:       vmID,
+		Domain:        strconv.FormatUint(sequence, 10),
```

Comment:
```suggestion
		Challenge:        strconv.FormatUint(sequence, 10),
```
It'd be clearer to use `Challenge` instead of `Domain` or `Nonce`: https://www.w3.org/TR/vc-data-model/#example-a-simple-example-of-a-verifiable-presentation.
The `Domain` is for the verifier (receiver) to specify the scope in which the `Challenge` value can have meaning. For example, the verifier can enforce the presenter to use a `Domain: did:panacea:abc...` and `Challenge: 10 (= sequence)`.
But now, I don't think we need to use `Domain`. It'd be enough to use `Challenge` for now. 

The `Nonce` is slightly different. In general, the `Challenge` is set by verifier (not presenter), as standardized as [Challenge-Response authentication](https://en.wikipedia.org/wiki/Challenge%E2%80%93response_authentication). In many cases, the presenter must set a response of the challenge to the `Challenge` field in the proof. But, in our case, the challenge response is the same as the challenge itself (it means that the presenter doesn't have to do any computation on the challenge).
On the other hand, the `Nonce` is usually chosen by presenter, for example, random values or time-based values. But, in practice, the uses of `Challenge` and `Nonce` are not clearly distinguished :)

It seems that it's preferred to use `Challenge` over `Nonce` these days: https://github.com/w3c/vc-data-integrity/issues/34, even though this link is not about the DID document proof.